core/src/main/java/org/elasticsearch/index/query/CommonTermsQueryParser.java

```diff
@@ -87,9 +87,6 @@ public QueryBuilder fromXContent(QueryParseContext parseContext) throws IOExcept
                         text = parser.objectText();
                     } else if ("analyzer".equals(currentFieldName)) {
                         analyzer = parser.text();
-                        if (parseContext.analysisService().analyzer(analyzer) == null) {
-                            throw new QueryParsingException(parseContext, "[common] analyzer [" + parser.text() + "] not found");
-                        }
```

Comment:
@javanna had to move this check to the query builder, will throw QueryShardException there now.